go-controller/pkg/ovn/pods_retry.go

```diff
@@ -0,0 +1,205 @@
+package ovn
+
+import (
+	"fmt"
+	"math/rand"
+	"net"
+	"time"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+	kapi "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/klog/v2"
+)
+
+// iterateRetryPods checks if any outstanding pods have been waiting for 60 seconds of last known failure
+// then tries to re-add them if so
+// updateAll forces all pods to be attempted to be retried regardless of the 1 minute delay
+func (oc *Controller) iterateRetryPods(updateAll bool) {
+	oc.retryPodsLock.Lock()
+	defer oc.retryPodsLock.Unlock()
+	now := time.Now()
+	for key, podEntry := range oc.retryPods {
+		if podEntry.ignore {
+			// neither addition nor deletion is being retried
+			continue
+		}
+
+		pod := podEntry.pod
+		podDesc := fmt.Sprintf("[%s/%s/%s]", pod.UID, pod.Namespace, pod.Name)
+		// it could be that the Pod got deleted, but Pod's DeleteFunc has not been called yet, so don't retry creation
+		if podEntry.needsAdd {
+			kPod, err := oc.watchFactory.GetPod(pod.Namespace, pod.Name)
+			if err != nil && errors.IsNotFound(err) {
+				klog.Infof("%s pod not found in the informers cache, not going to retry pod setup", podDesc)
+				podEntry.needsAdd = false
+			} else {
+				pod = kPod
+			}
+		}
+
+		if !util.PodScheduled(pod) {
+			klog.V(5).Infof("Retry: %s not scheduled", podDesc)
+			continue
+		}
+
+		podEntry.backoffSec = (podEntry.backoffSec * 2)
+		if podEntry.backoffSec > 60 {
+			podEntry.backoffSec = 60
+		}
+		backoff := (podEntry.backoffSec * time.Second) + (time.Duration(rand.Intn(500)) * time.Millisecond)
+		podTimer := podEntry.timeStamp.Add(backoff)
+		if updateAll || now.After(podTimer) {
+			// check if we need to retry delete first
+			if podEntry.needsDel != nil {
+				klog.Infof("%s retry pod teardown", podDesc)
+				if err := oc.removePod(pod, podEntry.needsDel); err == nil {
```

Comment:
nit you could just do err != nil here, then you dont need to do an else

Reply:
nice yeah fixed this.